packages/esm-patient-chart-app/src/banner/contact-details/contact-details.component.tsx

```diff
@@ -87,6 +92,66 @@ const Relationships: React.FC<{ patientId: string }> = ({ patientId }) => {
   );
 };
 
+const PatientLists: React.FC<{ patientId: string }> = ({ patientId }) => {
+  const { t } = useTranslation();
+  const { data: formattedPatientLists, isLoading } = usePatientLists(patientId);
+  const [showPatientListDetails, setShowPatientListDetails] = React.useState(false);
+  const togglePatientListDetails = React.useCallback((event: MouseEvent) => {
+    event.stopPropagation();
+    setShowPatientListDetails((value) => !value);
+  }, []);
+
+  return (
+    <>
+      <p className={styles.heading}>
+        {t('patientLists', 'Patient Lists')}({formattedPatientLists?.length})
+        <Button
+          kind="ghost"
+          renderIcon={(props) =>
+            showPatientListDetails ? <ChevronUp size={16} {...props} /> : <ChevronDown size={16} {...props} />
+          }
+          iconDescription="Toggle patient List Details"
+          onClick={togglePatientListDetails}
+          style={{ marginTop: '-0.25rem' }}
+        >
+          {showPatientListDetails ? t('seeLess', 'See less') : t('seeAll', 'See All')}
+        </Button>
+      </p>
+      <>
+        {(() => {
+          if (isLoading) return <SkeletonText />;
+          if (formattedPatientLists?.length && !showPatientListDetails) {
+            return (
+              <ul>
+                {formattedPatientLists.slice(0, 5).map((r) => (
+                  <li key={r.uuid} className={styles.relationship}>
+                    <ConfigurableLink className={styles.link} to={`\${openmrsSpaBase}/patient-list/${r.uuid}`}>
+                      {r.display}
+                    </ConfigurableLink>
+                  </li>
+                ))}
+              </ul>
+            );
+          } else if (formattedPatientLists?.length && showPatientListDetails) {
+            return (
+              <ul>
+                {formattedPatientLists.map((r) => (
+                  <li key={r.uuid} className={styles.relationship}>
+                    <ConfigurableLink className={styles.link} to={`\${openmrsSpaBase}/patient-list/${r.uuid}`}>
+                      {r.display}
+                    </ConfigurableLink>
+                  </li>
+                ))}
+              </ul>
+            );
+          }
+          return <p>--</p>;
+        })()}
```

Comment:
This looks okay to create a function and then calling it, but I believe this should not be done in the render. Instead, I believe we can save this whole logic using `useMemo` hook in a variable like:

```suggestion
        const patientLists = useMemo(() => {
          if (isLoading) return <SkeletonText />;
          if (formattedPatientLists?.length && !showPatientListDetails) {
            return (
              <ul>
                {formattedPatientLists.slice(0, 5).map((r) => (
                  <li key={r.uuid} className={styles.relationship}>
                    <ConfigurableLink className={styles.link} to={`\${openmrsSpaBase}/patient-list/${r.uuid}`}>
                      {r.display}
                    </ConfigurableLink>
                  </li>
                ))}
              </ul>
            );
          } else if (formattedPatientLists?.length && showPatientListDetails) {
            return (
              <ul>
                {formattedPatientLists.map((r) => (
                  <li key={r.uuid} className={styles.relationship}>
                    <ConfigurableLink className={styles.link} to={`\${openmrsSpaBase}/patient-list/${r.uuid}`}>
                      {r.display}
                    </ConfigurableLink>
                  </li>
                ))}
              </ul>
            );
          }
          return <p>--</p>;
        }, [<---dependencies--->])
        
        return (
        <div>
            <Header Code />
            {patientLists}
        </div>
```

Also, we haven't done any error handling, please add that too.
Thanks!